nifi-commons/nifi-flow-encryptor/src/main/java/org/apache/nifi/flow/encryptor/XmlFlowEncryptor.java

```diff
@@ -0,0 +1,103 @@
+import javax.xml.transform.stream.StreamSource;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Matcher;
+
+public class XmlFlowEncryptor extends AbstractFlowEncryptor {
+    private static final XMLEventReaderProvider eventReaderProvider = new StandardXMLEventReaderProvider();
+
+    @Override
+    public void processFlow(final InputStream inputStream, final OutputStream outputStream,
+                            final PropertyEncryptor inputEncryptor, final PropertyEncryptor outputEncryptor) {
+        XMLOutputFactory xmlOutputFactory = XMLOutputFactory.newInstance();
+        final XMLEventFactory eventFactory = XMLEventFactory.newInstance();
+        try {
+            final XMLEventReader reader = eventReaderProvider.getEventReader(new StreamSource(inputStream));
+            final XMLEventWriter writer = xmlOutputFactory.createXMLEventWriter(outputStream, StandardCharsets.UTF_8.name());
+            while (reader.hasNext()) {
+                final XMLEvent event = reader.nextEvent();
+                if (event.getEventType() == XMLEvent.START_DOCUMENT) {
+                    final StartDocument startDocument = (StartDocument) event;
+                    handleXmlStartDocument(startDocument, outputStream);
+                } else if (event.getEventType() == XMLEvent.CHARACTERS) {
+                    final Characters characters = event.asCharacters();
+                    final String value = characters.getData();
+                    final Matcher matcher = ENCRYPTED_PATTERN.matcher(value);
+                    if (matcher.matches()) {
+                        final String processedValue = getOutputEncrypted(matcher.group(FIRST_GROUP), inputEncryptor, outputEncryptor);
+                        writer.add(eventFactory.createCharacters(processedValue));
+                    } else {
+                        writer.add(characters);
+                    }
+                } else if (event.getEventType() == XMLEvent.COMMENT) {
+                    writer.add(event);
+                    final Characters newLine = eventFactory.createCharacters("\n");
+                    writer.add(newLine);
+                } else if (event.getEventType() == XMLEvent.END_DOCUMENT) {
+                    final Characters newLine = eventFactory.createCharacters("\n");
+                    writer.add(newLine);
+                } else {
+                    writer.add(event);
+                }
+            }
+            writer.flush();
+            writer.close();
+            reader.close();
+            outputStream.close();
+            inputStream.close();
+        } catch (XMLStreamException e) {
+            final String msg = String.format("XMLStreamException occurred - %s", e.getMessage());
+            throw new RuntimeException(msg, e);
```

Comment:
Instead of including and repeating the exception message, in this case it is better to have a simple message. The stack trace would include the message from the wrapped exception:
```suggestion
            throw new RuntimeException("Flow XML Processing Failed", e);
```